Core/Node-API/Include/Shared/napi/napi.h

```diff
@@ -56,2055 +72,3178 @@ static_assert(sizeof(char16_t) == sizeof(wchar_t), "Size mismatch between char16
-    operator float() const;    ///< Converts a Number value to a 32-bit floating-point value.
-    operator double() const;   ///< Converts a Number value to a 64-bit floating-point value.
-
-    int32_t Int32Value() const;   ///< Converts a Number value to a 32-bit signed integer value.
-    uint32_t Uint32Value() const; ///< Converts a Number value to a 32-bit unsigned integer value.
-    int64_t Int64Value() const;   ///< Converts a Number value to a 64-bit signed integer value.
-    float FloatValue() const;     ///< Converts a Number value to a 32-bit floating-point value.
-    double DoubleValue() const;   ///< Converts a Number value to a 64-bit floating-point value.
-  };
+  bool IsExceptionPending() const;
+  Error GetAndClearPendingException() const;
 
-// currently experimental guard with version of NAPI_VERSION that it is
-// released in once it is no longer experimental
-#if (NAPI_VERSION > 2147483646)
-  /// A JavaScript bigint value.
-  class BigInt : public Value {
-  public:
-    static BigInt New(
-      napi_env env, ///< N-API environment
-      int64_t value ///< Number value
-    );
-    static BigInt New(
-      napi_env env,  ///< N-API environment
-      uint64_t value ///< Number value
-    );
-
-    /// Creates a new BigInt object using a specified sign bit and a
-    /// specified list of digits/words.
-    /// The resulting number is calculated as:
-    /// (-1)^sign_bit * (words[0] * (2^64)^0 + words[1] * (2^64)^1 + ...)
-    static BigInt New(
-      napi_env env,          ///< N-API environment
-      int sign_bit,          ///< Sign bit. 1 if negative.
-      size_t word_count,     ///< Number of words in array
-      const uint64_t* words  ///< Array of words
-    );
-
-    BigInt();                               ///< Creates a new _empty_ BigInt instance.
-    BigInt(napi_env env, napi_value value); ///< Wraps a N-API value primitive.
-
-    int64_t Int64Value(bool* lossless) const;   ///< Converts a BigInt value to a 64-bit signed integer value.
-    uint64_t Uint64Value(bool* lossless) const; ///< Converts a BigInt value to a 64-bit unsigned integer value.
-
-    size_t WordCount() const; ///< The number of 64-bit words needed to store the result of ToWords().
-
-    /// Writes the contents of this BigInt to a specified memory location.
-    /// `sign_bit` must be provided and will be set to 1 if this BigInt is negative.
-    /// `*word_count` has to be initialized to the length of the `words` array.
-    /// Upon return, it will be set to the actual number of words that would
-    /// be needed to store this BigInt (i.e. the return value of `WordCount()`).
-    void ToWords(int* sign_bit, size_t* word_count, uint64_t* words);
-  };
-#endif  // NAPI_EXPERIMENTAL
+  MaybeOrValue<Value> RunScript(const char* utf8script) const;
+  MaybeOrValue<Value> RunScript(const std::string& utf8script) const;
+  MaybeOrValue<Value> RunScript(String script) const;
```

Comment:
This is new. It can maybe replace the Eval function we have? It just needs the source parameter.

Reply:
not really, `RunSript` calls `napi_run_script` with empty `source_url`.

Reply:
No need to block the PR for this, just asking. Maybe file an issue.

Reply:
Ho! I see! you mean replace the eval code in env.cc ?

Reply:
I did the change.

Reply:
This is good. Maybe even remove the Eval function completely, but then it will be breaking, so maybe not. :-)